commit 2162801457399b3efeab96a9454825f36be60e48 (refs/changes/78/2478/1)
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2017-08-30T07:05:22-06:00 (2 years, 1 month ago)
    
    MANATEE-303 postgresql needs to be configured with tcp keepalive or equivalent

diff --git a/etc/9.2/postgresql.manta.coal.conf b/etc/9.2/postgresql.manta.coal.conf
index 8218750..de4b415 100644
--- a/etc/9.2/postgresql.manta.coal.conf
+++ b/etc/9.2/postgresql.manta.coal.conf
@@ -96,12 +96,12 @@ max_connections = 100     # (change requires restart)
 # - TCP Keepalives -
 # see "man 7 tcp" for details
 
-#tcp_keepalives_idle = 0    # TCP_KEEPIDLE, in seconds;
-          # 0 selects the system default
-#tcp_keepalives_interval = 0    # TCP_KEEPINTVL, in seconds;
-          # 0 selects the system default
-#tcp_keepalives_count = 0   # TCP_KEEPCNT;
-          # 0 selects the system default
+tcp_keepalives_idle = 15                # TCP_KEEPIDLE, in seconds;
+                                        # 0 selects the system default
+tcp_keepalives_interval = 3             # TCP_KEEPINTVL, in seconds;
+                                        # 0 selects the system default
+tcp_keepalives_count = 5                # TCP_KEEPCNT;
+                                        # 0 selects the system default
 
 
 #------------------------------------------------------------------------------
diff --git a/etc/9.2/postgresql.manta.lab.conf b/etc/9.2/postgresql.manta.lab.conf
index d330a36..4b0e9f8 100644
--- a/etc/9.2/postgresql.manta.lab.conf
+++ b/etc/9.2/postgresql.manta.lab.conf
@@ -96,12 +96,12 @@ superuser_reserved_connections = 10 # (change requires restart)
 # - TCP Keepalives -
 # see "man 7 tcp" for details
 
-#tcp_keepalives_idle = 0    # TCP_KEEPIDLE, in seconds;
-          # 0 selects the system default
-#tcp_keepalives_interval = 0    # TCP_KEEPINTVL, in seconds;
-          # 0 selects the system default
-#tcp_keepalives_count = 0   # TCP_KEEPCNT;
-          # 0 selects the system default
+tcp_keepalives_idle = 15                # TCP_KEEPIDLE, in seconds;
+                                        # 0 selects the system default
+tcp_keepalives_interval = 3             # TCP_KEEPINTVL, in seconds;
+                                        # 0 selects the system default
+tcp_keepalives_count = 5                # TCP_KEEPCNT;
+                                        # 0 selects the system default
 
 
 #------------------------------------------------------------------------------
diff --git a/etc/9.2/postgresql.manta.production.conf b/etc/9.2/postgresql.manta.production.conf
index 2f5c6d1..8c9ccbd 100644
--- a/etc/9.2/postgresql.manta.production.conf
+++ b/etc/9.2/postgresql.manta.production.conf
@@ -96,12 +96,12 @@ max_connections = 1000     # (change requires restart)
 # - TCP Keepalives -
 # see "man 7 tcp" for details
 
-#tcp_keepalives_idle = 0    # TCP_KEEPIDLE, in seconds;
-          # 0 selects the system default
-#tcp_keepalives_interval = 0    # TCP_KEEPINTVL, in seconds;
-          # 0 selects the system default
-#tcp_keepalives_count = 0   # TCP_KEEPCNT;
-          # 0 selects the system default
+tcp_keepalives_idle = 15                # TCP_KEEPIDLE, in seconds;
+                                        # 0 selects the system default
+tcp_keepalives_interval = 3             # TCP_KEEPINTVL, in seconds;
+                                        # 0 selects the system default
+tcp_keepalives_count = 5                # TCP_KEEPCNT;
+                                        # 0 selects the system default
 
 
 #------------------------------------------------------------------------------
diff --git a/etc/9.6/postgresql.manta.coal.conf b/etc/9.6/postgresql.manta.coal.conf
index 394f354..5170bb8 100644
--- a/etc/9.6/postgresql.manta.coal.conf
+++ b/etc/9.6/postgresql.manta.coal.conf
@@ -97,11 +97,11 @@ max_connections = 100                   # (change requires restart)
 # - TCP Keepalives -
 # see "man 7 tcp" for details
 
-#tcp_keepalives_idle = 0                # TCP_KEEPIDLE, in seconds;
+tcp_keepalives_idle = 15                # TCP_KEEPIDLE, in seconds;
                                         # 0 selects the system default
-#tcp_keepalives_interval = 0            # TCP_KEEPINTVL, in seconds;
+tcp_keepalives_interval = 3             # TCP_KEEPINTVL, in seconds;
                                         # 0 selects the system default
-#tcp_keepalives_count = 0               # TCP_KEEPCNT;
+tcp_keepalives_count = 5                # TCP_KEEPCNT;
                                         # 0 selects the system default
 
 
diff --git a/etc/9.6/postgresql.manta.lab.conf b/etc/9.6/postgresql.manta.lab.conf
index 3d844fc..9bd60ee 100644
--- a/etc/9.6/postgresql.manta.lab.conf
+++ b/etc/9.6/postgresql.manta.lab.conf
@@ -97,11 +97,11 @@ superuser_reserved_connections = 10     # (change requires restart)
 # - TCP Keepalives -
 # see "man 7 tcp" for details
 
-#tcp_keepalives_idle = 0                # TCP_KEEPIDLE, in seconds;
+tcp_keepalives_idle = 15                # TCP_KEEPIDLE, in seconds;
                                         # 0 selects the system default
-#tcp_keepalives_interval = 0            # TCP_KEEPINTVL, in seconds;
+tcp_keepalives_interval = 3             # TCP_KEEPINTVL, in seconds;
                                         # 0 selects the system default
-#tcp_keepalives_count = 0               # TCP_KEEPCNT;
+tcp_keepalives_count = 5                # TCP_KEEPCNT;
                                         # 0 selects the system default
 
 
diff --git a/etc/9.6/postgresql.manta.production.conf b/etc/9.6/postgresql.manta.production.conf
index f47d134..10f800d 100644
--- a/etc/9.6/postgresql.manta.production.conf
+++ b/etc/9.6/postgresql.manta.production.conf
@@ -97,11 +97,11 @@ max_connections = 1000                  # (change requires restart)
 # - TCP Keepalives -
 # see "man 7 tcp" for details
 
-#tcp_keepalives_idle = 0                # TCP_KEEPIDLE, in seconds;
+tcp_keepalives_idle = 15                # TCP_KEEPIDLE, in seconds;
                                         # 0 selects the system default
-#tcp_keepalives_interval = 0            # TCP_KEEPINTVL, in seconds;
+tcp_keepalives_interval = 3             # TCP_KEEPINTVL, in seconds;
                                         # 0 selects the system default
-#tcp_keepalives_count = 0               # TCP_KEEPCNT;
+tcp_keepalives_count = 5                # TCP_KEEPCNT;
                                         # 0 selects the system default
 
 
